Makes gauze/ointment craftable; refactors a bit of crafting code

JimKil3 opened the pull request:
<!-- By ticking or leaving ticked the option "Allow edits and access to secrets by maintainers" you give permission for repository maintainers to push changes to your branch without explicitly asking. -->

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## What Does This PR Do
Title. Improvised gauze (from cloth) can be treated with medicine and made into normal gauze (which heals damages on application), and ointment can be crafted with plastic and some reagents.
### Gauze Recipe
- 6 Improvised Gauze
- 30u Styptic Powder
- 10u Sterilizine
... makes 6 normal gauze

### Ointment Recipe
- 2 Plastic Sheets
- 10u Water
- 30u Silver Sulfadiazine
- 10u Sterilizine
... makes 6 ointment

The PR also refactors crafting code to properly work with multiple reagents and allows `/obj/item/stack/medical` to be put in medical smartfridges.
<!-- Include a small to medium description of what your PR changes. -->
<!-- Document all changes, as not doing this may delay reviews or even discourage maintainers from merging your PR! -->
<!-- If your PR fixes an issue, add "Fixes #1234" somewhere in the PR description. This will automatically close the bug upon PR submission. -->

## Why It's Good For The Game
Giving medchem more to make than patches and pills is good, and gauze/ointment are useful alternatives to straight styptic powder/silver sulfadiazine patches. Sterilizine is also a chemical that doesn't see much use, and this application makes sense for it (to sterilize the cloth and to sterilize burn wounds).
<!-- Add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Testing
Loaded in, made gauze, made ointment, tried various combinations of beakers and ensured they all drained properly, crafted FRAG-12
<!-- How did you test the PR, if at all? -->

## Changelog
:cl:
add: You can now craft (non-improvised) gauze and ointment!
add: All medical stacks can now be added to medical smartfridges; gauze, ointment, adv. trauma kits, etc
fix: Fixed some peculiar behavior in recipes with multiple reagent components
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->
<!-- If a PR has no impact on players (i.e. a code refactor that does not change functionality) then the entire Changelog heading and contents can be removed. -->

Zorazi-Teo-Daedalus commented:
Take it with a grain of salt because I'm not a balance team member or anything, but I do play a lot of medchem lately and I've been a long time medical player in general, so I wanted to give my two cents.

This is cool, and I like it, but I think it would be better served as crafting the advanced trauma kits rather than just gauze and ointments. The reason I feel that way is because gauze and ointment is typically unused by medical staff, but trauma kits actually see use. So the time investiture (which is admittedly not a massive amount) into making gauze or ointment could've just gone into mass producing patches or refilling menders, which see a lot more use than gauze or ointment. 

TL;DR advanced trauma kits would be better because it's something doctors could conceivably *want* from medchem alongside things like patches and mender refills. Otherwise, great PR and I love the spirit of it and the direction you're trying to go with it.

JimKil3 commented:
>TL;DR advanced trauma kits would be better because it's something doctors could conceivably want from medchem alongside things like patches and mender refills. Otherwise, great PR and I love the spirit of it and the direction you're trying to go with it.

Advanced trauma kits heal more damage, yes, but gauze stops bleeding instantly. The crafting recipe for gauze also doubles the efficiency of the styptic powder used (30u -> 6 items that heal 10 each, meanwhile 30u splashed would only normally heal 30 brute). I might add a crafting recipe to advanced trauma/burn kits in the future, but they'll have a much harsher crafting recipe, and the idea with this PR is to give medchem something to make other than just pills/patches, not necessarily to supply doctors with new tools (gauze is available in nanomeds, after all). In addiction, chemistry could supply the general population with gauze/ointment, making the doctor's lives easier in a different way.

SchrodingersWolf commented:
I do echo the sentiment of Teo - Generally speaking gauze and ointment's more convoluted application method makes them undesirable for doctors to use, and brute/burn kits only gets used sometimes because of how much damage it heals and the overheal to other limbs mechanic. (I am aware that gauze/ointment also shares this mechanic, but 10 points of healing generally is not enough to spread it to other limbs.)

Not to mention that gauze/ointment are abundant enough that I feel like MedChem won't go the extra mile to make them, and normal crew generally prefers patches/pills since they do not incur a timer when used on themselves.

That being said, I don't see a reason why these recipes can't be added, though could you make it so the Chemistry fridges also accept gauze/ointments? Most of what MedChem can make are allowed inside the fridge, and gauze/ointments being excluded from the fridge would be quite unintuitive.

JimKil3 commented:
>"...and brute/burn kits only gets used sometimes because of how much damage it heals and the overheal to other limbs mechanic..."

This is just incorrect. Both I and other medical / normal players I know use gauze to stop bleeding, and ointment can be used for fixing critical burns. Gauze / ointment also can be faster than menders to use if the doctor knows what they're doing & the damage is localized to one limb.

>"Not to mention that gauze/ointment are abundant enough that I feel like MedChem won't go the extra mile to make them, and normal crew generally prefers patches/pills since they do not incur a timer when used on themselves."

Doctors tend to use automenders over gauze / ointment, yes. Chemistry making gauze / ointment, I imagine, will fall into a similar niche to making things like salbutamol or epinephrine - not every chemist will be making it, but it's still additional (and interesting!) content for chemistry players. As for normal crew, while yes, patches and pills have no timer, gauze/ointment are simpler to understand, more effective (in most cases, normal crew will either not have much damage or just go to medbay and be treated by a doctor), and easier to store.

>"though could you make it so the Chemistry fridges also accept gauze/ointments?"

Yes, I added this and updated the PR desc w/ commits [a422df2](https://github.com/ParadiseSS13/Paradise/pull/23127/commits/a422df20c3a1f7d169d9ac489210e78f548fd427) and [94198dd](https://github.com/ParadiseSS13/Paradise/pull/23127/commits/94198dd6a6f80613aab3a35b0be7251894004adc).

Henri215 commented:
> > "...and brute/burn kits only gets used sometimes because of how much damage it heals and the overheal to other limbs mechanic..."
> 
> This is just incorrect. Both I and other medical / normal players I know use gauze to stop bleeding, and ointment can be used for fixing critical burns. Gauze / ointment also can be faster than menders to use if the doctor knows what they're doing & the damage is localized to one limb.
> 
> > "Not to mention that gauze/ointment are abundant enough that I feel like MedChem won't go the extra mile to make them, and normal crew generally prefers patches/pills since they do not incur a timer when used on themselves."
> 
> Doctors tend to use automenders over gauze / ointment, yes. Chemistry making gauze / ointment, I imagine, will fall into a similar niche to making things like salbutamol or epinephrine - not every chemist will be making it, but it's still additional (and interesting!) content for chemistry players. As for normal crew, while yes, patches and pills have no timer, gauze/ointment are simpler to understand, more effective (in most cases, normal crew will either not have much damage or just go to medbay and be treated by a doctor), and easier to store.
> 
> > "though could you make it so the Chemistry fridges also accept gauze/ointments?"
> 
> Yes, I added this and updated the PR desc w/ commits [a422df2](https://github.com/ParadiseSS13/Paradise/pull/23127/commits/a422df20c3a1f7d169d9ac489210e78f548fd427) and [94198dd](https://github.com/ParadiseSS13/Paradise/pull/23127/commits/94198dd6a6f80613aab3a35b0be7251894004adc).

Worth adding to the changelog that fridges can receive gauza/ointments now.

JimKil3 commented:
> Worth adding to the changelog that fridges can receive gauza/ointments now.
Done

DGamerL commented on code/modules/crafting/recipes.dm:
```diff
@@ -1352,3 +1352,22 @@
 	subcategory = CAT_LARGE_DECORATIONS
+
+/datum/crafting_recipe/gauze
+	name = "Treated Gauze"
```
```suggestion
	name = "Treated gauze"
```

DGamerL reviewed (commented):
Get nitpicked

DGamerL reviewed (approved):
list(list(list(L), list(G)), list(T), list(M))
